components/tikv_util/src/logger/file_log.rs

```diff
@@ -133,63 +148,157 @@ impl Drop for RotatingFileLogger {
     }
 }
 
+pub struct RotateByTime {
+    rotation_timespan: Duration,
+    next_rotation_time: DateTime<Utc>,
+}
+
+impl RotateByTime {
+    pub fn new(rotation_timespan: Duration) -> Self {
+        Self {
+            rotation_timespan,
+            next_rotation_time: compute_rotation_time(&Utc::now(), rotation_timespan),
+        }
+    }
+
+    #[cfg(test)]
+    pub fn new_for_test(next_rotation_time: DateTime<Utc>, rotation_timespan: Duration) -> Self {
+        Self {
+            next_rotation_time,
+            rotation_timespan,
+        }
+    }
+}
+
+impl Rotator for RotateByTime {
+    fn should_rotate(&self, _: &File) -> io::Result<bool> {
+        Ok(Utc::now() > self.next_rotation_time)
+    }
+
+    fn rotate(&mut self, path: &Path, file: &mut File) -> io::Result<Option<PathBuf>> {
+        file.flush()?;
+
+        let renamed = match rename_with_timestamp(path) {
+            Ok(path) => Some(path),
+            Err(ref e) if e.kind() == ErrorKind::NotFound => None,
+            Err(e) => return Err(e),
+        };
+
+        self.next_rotation_time = compute_rotation_time(&Utc::now(), self.rotation_timespan);
+
+        *file = open_log_file(&path)?;
+
+        Ok(renamed)
+    }
+
+    fn validate(&self) -> bool {
+        !self.rotation_timespan.is_zero()
+    }
+}
+
+pub struct RotateBySize {
+    rotation_size: u64,
+}
+
+impl RotateBySize {
+    pub fn new(rotation_size: u64) -> Self {
+        RotateBySize { rotation_size }
```

Comment:
Why not using `ReadableSize` here

Reply:
Because for `rotation_timespan` is `ReadableDuration` but transform into `chrono::Duration` while handle by a `Rotator`, same does the old code.

Reply:
I think the implication of the `MB`, the unit of `rotation_size`,  is not as clear as `ReadableSize`, while DateTime does have clear units.